### DISCUSSION TITLE
Improvements to Collections - Trip Planning Workflow

### DISCUSSION THREAD

**Discussion by radh21301** [seanmorley15/AdventureLog#304] [Ideas]
I will try my best to explain what I think, and I hope this makes some sense. Inspirations from a pretty cool app called [Furkot](https://trips.furkot.com/).

**The difficulties I currently face:**
The way I see it, to plan an trip, we add a bunch of points (adventures), decide to spend a certain amount of time at that adventure, and then travel from one adventure to another.

Also, We wont really have the sequence of adventures planned out. We might have a rough idea, but nothing is set in stone.

In the transportation card, we have the ability to add the start and end locations. I think this can be improved in the sense that we should be able to link  adventures for the start and end locations. The way I plan it, transportation is always going to be from one adventure to another. Right now, I need to add a start date/time, end date/time, start location, end location etc..which is a lot of information and is not streamlined?

**Solution?:**

* Create a collection and give it a 'Start Date&Time' and 'End Date &Time'
* This should create a calendar like view
* Add a bunch of adventures to the collection.
  * Assign a day number instead of date: In my opinion, people plan things based on the day of the trip. I mean, I usually decide to do places A,B and C on Day 1..then M and N on Day 2 and so on. So, instead of assigning Dates to an adventure, we need to assign a Day number. The adventure has would then have the following properties 'Day number', 'Start time', and ''Duration'. No End Day/time -> The duration would take care of this.
  * Note: The assignments here are just starting points. We will have the ability to easily move them around as explained below.
* Once you have all your adventures assigned to days, the app should show a timeline/calendar like view.
  * Just like it is now, but with some extra capabilities.
  * We should be able to move adventures by dragging and dropping them on the timeline. As they are moved around the 'Day/Time' should change to match the calendar/timeline view.

*Transportation*
* The transportation cards or information should be automatically determined between all adventures. I mean, we have the start and end locations from the adventures -> so we can use Open street maps or something to get the time and distance to travel between those points - based on the mode of travel (bike, car etc..)
* I think this workflow would provide the best experience. If we don't have this feature: Since we move adventures around, we need to use google maps or something to get the duration between points for each change we make. It will end up becoming a headache, and I would eventually stop adding transportation cards.

Look at this example of how everything comes together:
  * Start of the day. Adventure 1 (Hiking): 9 AM. Duration is 2 hrs
     * The transportation info is gathered from OSM using Adventure 1 (Hiking) and Adventure 2 (Museum). Say this is 4 hrs.
  * Adventure 2 (Museum). This will then start at 9 AM + 2 hrs (duration in museum) + 4 hrs (travel) = 1 PM
  * The rest of the day is populated similarly
  * Now if I I move Adventure 2 above adventure 1, then everything is recalculated.
  * Hopefully this makes sense

**An icing on the cake feature:**
Say the collections page also has a map view of all the points, along with the transportation between them - > basically shows the sequence in which we are going to be doing the trip -> That would be so cool! This is also very useful to optimize a trip. When we look at the points on a map, we can then reorder the adventures so we cover all the locations that are close by around the same day/time.

*For some inspiration, here is screenshot of how a trip is planned using Furkot:*
<img width="1506" alt="image" src="https://github.com/user-attachments/assets/fcf2707c-0aae-4a52-9a80-7a0112561ad0">

Thank you for all the work you do!

**Comment by lolo9538** [comment]
That looks cool. However I think was you describe is more of a trip planner whereas this app was designed as a journal I think. In the end it's true we want to combine both. Start with a trip planner that will become your trip journal...

**Comment by joeshmoe57** [comment]
+1 on "day number planning" option. I just set up AdventureLog and tried it out for a road trip (love it, thank you!), but we ended up starting one day after I initially planned. So adjusting all dates by one is tedious. Would have been next to no effort to choose a different start date.
